### PR TITLE
User Story 60501: aci-linuxcontainer-public-ip

### PR DESCRIPTION
**Pull request by TomArcherMsft**
Part of POC to test generating sample code and articles using OpenAI.

### CODE REVIEW

File: quickstart/101-aci-linuxcontainer-public-ip/main.tf [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  prefix = var.resource_group_name_prefix
+}
+
+resource "azurerm_resource_group" "rg" {
+  name     = random_pet.rg_name.id
+  location = var.resource_group_location
+}
+
+resource "azurerm_container_group" "container" {
+  name                = var.name
```
Comment (grayzu) [inline quickstart/101-aci-linuxcontainer-public-ip/main.tf]: This variable should be more specific about what the name is for.

File: quickstart/101-aci-linuxcontainer-public-ip/providers.tf [hunk 2]
```diff
@@ -0,0 +1,16 @@
+terraform {
+  required_version = ">=0.12"
```
Comment (grayzu) [inline quickstart/101-aci-linuxcontainer-public-ip/providers.tf]: This should be more current.

File: quickstart/101-aci-linuxcontainer-public-ip/providers.tf [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~>2.0"
```
Comment (grayzu) [inline quickstart/101-aci-linuxcontainer-public-ip/providers.tf]: Is there a reason that this is 2.0? If not, it should be updated to a more current version.

File: quickstart/101-aci-linuxcontainer-public-ip/variables.tf [hunk 4]
```diff
@@ -0,0 +1,44 @@
+variable "resource_group_location" {
+  default     = "eastus"
```
Comment (grayzu) [inline quickstart/101-aci-linuxcontainer-public-ip/variables.tf:3]: Add types to the variables.

File: quickstart/101-aci-linuxcontainer-public-ip/variables.tf [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  description = "Location for all resources."
+}
+
+variable "resource_group_name_prefix" {
+  default     = "rg"
+  description = "Prefix of the resource group name that's combined with a random ID so name is unique in your Azure subscription."
+}
+
+variable "name" {
+  description = "Name for the container group"
```
Comment (grayzu) [inline quickstart/101-aci-linuxcontainer-public-ip/variables.tf]: This is actually the name for the container group and the container, correct?

**Review by grayzu — CHANGES_REQUESTED** [review]
Thanks! Looking good. Few comments.

File: quickstart/101-aci-linuxcontainer-public-ip/main.tf (hunk 1 above)
Comment (TomArcherMsft) [reply quickstart/101-aci-linuxcontainer-public-ip/main.tf]: I'm not sure what name would be more appropriate here.

According to the Bicep version of this article, "In this quickstart, you use a Bicep file to deploy an isolated Docker container and make its web application available with a public IP address." 

What variable name do you think is more appropriate?

File: quickstart/101-aci-linuxcontainer-public-ip/providers.tf (hunk 2 above)
Comment (TomArcherMsft) [reply quickstart/101-aci-linuxcontainer-public-ip/providers.tf]: Changed.

File: quickstart/101-aci-linuxcontainer-public-ip/providers.tf (hunk 3 above)
Comment (TomArcherMsft) [reply quickstart/101-aci-linuxcontainer-public-ip/providers.tf]: Changed.

**Comment by TomArcherMsft** [comment]
@lonegunmanb @grayzu This is ready for review.

**Pull request closed without merging**